doc/developer/design/20231211_datadog_sink.md

```diff
@@ -0,0 +1,162 @@
+# Datadog Sink
+
+- Associated issues:
+    - https://github.com/MaterializeInc/materialize/issues/17601
+    - https://github.com/MaterializeInc/materialize/issues/4779
+
+## The Problem
+
+Users want to export **metrics** to Datadog. But they need to learn how to configure, test, deploy, and maintain two external services: the Datadog Agent and a SQL Exporter — making the experience long and sometimes even frustrating.
+
+## Success Criteria
+
+In a few clicks, a user can integrate Materialize with Datadog. Also, the user can export custom metrics to Datadog by running the following SQL commands:
+
+```sql
+CREATE CONNECTION conn_datadog TO DATADOG API KEY = ...;
+CREATE SINK FROM <view/table> TO DATADOG CONNECTION conn_datadog;
+```
+
+The source content of the sink, which can be either a view or a table, will be defined by the user but needs to match a predefined structure.
+
+## Out of Scope
+
+It is out of the scope:
+
+- Alternative approaches, such as https://github.com/MaterializeInc/materialize/issues/4779, to select labels and values, or send any additional information to Datadog outside the connection validation or metrics.
+- Compression algorithms before exporting metrics.
+
+## Solution Proposal
+
+The solution proposal involves a new connection and sink for Datadog, and an OAuth process to create a seamless experience.
+
+### Datadog Connection
+
+The new connection will store Datadog’s API key, which is necessary to submit metrics to Datadog. The first time the user runs the integration, an API key will be created automatically by the console (refer to the [OAuth process diagram](#oauth-process) for more details.)
```

Comment:
I'll drop some more detailed comments down below, but for v1 I think we should just have the user create an API key manually and install it in Materialize as they would any secret:

```sql
CREATE SECRET datadog_api_key AS ...
```

That will be 100x easier than dealing with OAuth out of the gate.